### PR TITLE
Handle `hf://` urls + raise `ValueError` if repo type is unknown

### PR DESCRIPTION
**Pull request by Wauplin**
Related to [hffs](https://github.com/huggingface/hffs) library (see https://github.com/huggingface/hffs/pull/10#discussion_r1073546356) cc @mariosasko. It feels cleaner if `huggingface_hub` can directly parse urls starting by `"hf://"` (and will surely avoid some troubles in the future).

PR includes:
- Support for `RepoUrl("hf://datasets/my-user/my-dataset")`
- `repo_type_and_id_from_hf_id` now raises a ValueError if `repo_type` is unknown. Avoid to silently consider "https://huggingface.co/spaesc/mdnestor/tortoise" as a `model` instead of a `space` just because of the typo.
- Updated docstrings + added some tests

### CODE REVIEW

**Review by mariosasko — APPROVED** [review]
Thanks for the speedy PR 🙂. LGTM!

File: src/huggingface_hub/hf_api.py [hunk 1]
```diff
@@ -112,9 +114,21 @@ def repo_type_and_id_from_hf_id(
     Returns:
         A tuple with three items: repo_type (`str` or `None`), namespace (`str` or
         `None`) and repo_id (`str`).
+
+    Raises:
+        - [`ValueError`](https://docs.python.org/3/library/exceptions.html#ValueError)
+            If URL cannot be parsed.
+        - [`ValueError`](https://docs.python.org/3/library/exceptions.html#ValueError)
+            If `repo_type` is unknown.
```
Comment (LysandreJik) [inline src/huggingface_hub/hf_api.py:122]: Ah that's cool to put a link to the docs of the errors

File: src/huggingface_hub/hf_api.py [hunk 2]
```diff
@@ -112,9 +114,21 @@ def repo_type_and_id_from_hf_id(
...
     """
+    input_hf_id = hf_id
     hub_url = re.sub(r"https?://", "", hub_url if hub_url is not None else ENDPOINT)
     is_hf_url = hub_url in hf_id and "@" not in hf_id
+
+    HFFS_PREFIX = "hf://"
+    if hf_id.startswith(HFFS_PREFIX):  # Remove "hf://" prefix if exists
+        hf_id = hf_id[len(HFFS_PREFIX) :]
```
Comment (LysandreJik) [inline src/huggingface_hub/hf_api.py:130]: :+1:

**Review by LysandreJik — APPROVED** [review]
LGTM

**Pull request merged**